pkg/config/analysis/local/istiod_analyze.go

```diff
@@ -276,15 +280,20 @@ func (sa *IstiodAnalyzer) AddRunningKubeSourceWithRevision(c kubelib.Client, rev
 		return
 	}
 	sa.stores = append(sa.stores, store)
-	err = store.SetWatchErrorHandler(func(r *cache.Reflector, err error) {
-		// failed resources will never be synced, which causes the process to hang indefinitely.
-		// better to fail fast, and get a good idea for the failure.
-		scope.Analysis.Errorf("Failed to watch crd resource for analysis: %s", err)
-	})
-	if err != nil {
-		scope.Analysis.Errorf("error setting up error handling for kube crdclient: %v", err)
-		return
+	if !store.HasStarted() {
+		err = store.SetWatchErrorHandler(func(r *cache.Reflector, err error) {
```

Comment:
Looks this will make some informer having no watch error handler, not a big deal.

Just wondering if we could check HasStarted() for each informer

Reply:
In an real world, this will not happen in istioctl, istioctl always run with new client.
For istio-operator, all informers started in loop-0, when users modify the iop configuration, re-set watch error handler will cause error.